src/target/riscv/riscv.c

```diff
@@ -4521,6 +4521,23 @@ COMMAND_HANDLER(riscv_set_maskisr)
 	return ERROR_OK;
 }
 
+COMMAND_HANDLER(riscv_set_autofence)
+{
+	struct target *target = get_current_target(CMD_CTX);
+	RISCV_INFO(r);
+
+	if (CMD_ARGC != 1) {
+		command_print(CMD, "autofence enabled: %s", r->autofence ? "on" : "off");
+		return ERROR_OK;
+	} else if (CMD_ARGC == 1) {
+		COMMAND_PARSE_ON_OFF(CMD_ARGV[0], r->autofence);
+		return ERROR_OK;
+	}
+
+	LOG_ERROR("Command takes 0 or 1 parameters");
+	return ERROR_COMMAND_SYNTAX_ERROR;
```

Comment:
1) There seems to be mistake in this function - the last two lines of the function are currently dead code.

2) Per current OpenOCD conventions, LOG_ERROR should not be printed when ERROR_COMMAND_SYNTAX_ERROR is returned. OpenOCD will print the command syntax to the user on its own.

3) It is more natural to say simply `autofence: off` instead of `autofence enabled: off`. 

```suggestion
	if (CMD_ARGC == 0) {
		command_print(CMD, "autofence: %s", r->autofence ? "on" : "off");
		return ERROR_OK;
	} else if (CMD_ARGC == 1) {
		COMMAND_PARSE_ON_OFF(CMD_ARGV[0], r->autofence);
		return ERROR_OK;
	}
	return ERROR_COMMAND_SYNTAX_ERROR;
```

Reply:
My mistake, addressed

Reply:
> LOG_ERROR should not be printed when ERROR_COMMAND_SYNTAX_ERROR is returned.

Another [PR](https://github.com/riscv-collab/riscv-openocd/pull/1160 ) is created to remove unnecessary LOG_ERROR